server/internal/externalmcp/registryclient.go

```diff
@@ -163,3 +169,81 @@ func (c *RegistryClient) ListServers(ctx context.Context, registry Registry, par
 
 	return servers, nil
 }
+
+// ServerDetails contains detailed information about an MCP server including connection info.
+type ServerDetails struct {
+	Name        string
+	Description string
+	Version     string
+	RemoteURL   string
+}
+
+// getServerResponse wraps a single server from the registry.
+type getServerResponse struct {
+	Server serverJSON `json:"server"`
+}
+
+// GetServerDetails fetches server details including the remote URL from the registry.
+func (c *RegistryClient) GetServerDetails(ctx context.Context, registry Registry, serverName string) (*ServerDetails, error) {
+	u, err := url.Parse(registry.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse registry URL: %w", err)
+	}
+	u = u.JoinPath("v0.1", "servers", url.PathEscape(serverName), "versions", "latest")
+
+	c.logger.InfoContext(ctx, "fetching server details from registry", attr.SlogURL(u.String()))
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	for key, values := range checkForPulseCredentials(registry.URL) {
+		for _, value := range values {
+			req.Header.Set(key, value)
+		}
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch from registry: %w", err)
+	}
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("registry returned status %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var serverResp getServerResponse
+	if err := json.Unmarshal(body, &serverResp); err != nil {
+		return nil, fmt.Errorf("failed to decode server response: %w", err)
+	}
+
+	// Find the remote URL, preferring streamable-http over sse
+	var remoteURL string
+	for _, remote := range serverResp.Server.Remotes {
+		if remote.Type == "streamable-http" {
+			remoteURL = remote.URL
+			break
+		}
+		if remote.Type == "sse" && remoteURL == "" {
+			remoteURL = remote.URL
+		}
```

Comment:
Are we going to support calling out to SSE MCP servers? Can we write these off and only support servers using `streamable-http`? It's a deprecated transport after all.